src/tests/read_test.cpp

```diff
@@ -0,0 +1,75 @@
+
+#define BOOST_TEST_MODULE ASTE
+#define BOOST_TEST_DYN_LINK
+#include <boost/test/unit_test.hpp>
```

Comment:
This should be located in a separate file. Let's call it `testing.cpp`.

In CMake, you can then do:
```
add_executable(tests testing.cpp read_tests.cpp write_tests.cpp mesh.cpp)
```
Which links all tests together into a single binary.